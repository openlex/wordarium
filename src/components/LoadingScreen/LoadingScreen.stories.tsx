import React from "react";
import { LoadingScreen } from "@components";

export default {
	title: "Экран загрузки",
	components: LoadingScreen,
};

export const defaultLoadingScreen = () => <LoadingScreen />;
