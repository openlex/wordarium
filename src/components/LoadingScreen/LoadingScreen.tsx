import React from "react";
import { ProgressCircle } from "@components";
import styled from "@emotion/styled";
import { mixinFlexCenter } from "@styles";

const Wrapper = styled.div`
    ${mixinFlexCenter}
    flex-direction: column;
    width: 100%;
    height: 80vh;
`;
const Text = styled.div`
    margin-bottom: 5px;
`;

export const LoadingScreen = () => {
    return (
        <Wrapper>
            <Text>Loading</Text>
            <ProgressCircle size={20} time={2} strokeWidth={2} />
        </Wrapper>
    );
};
