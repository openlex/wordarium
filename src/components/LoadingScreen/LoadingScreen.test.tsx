import React from "react";
import { LoadingScreen } from "@components/LoadingScreen/LoadingScreen";
import { shallow } from "enzyme";

describe("loading screen", () => {
    it("markup", () => {
        expect(shallow(<LoadingScreen />)).toMatchSnapshot();
    });
});
