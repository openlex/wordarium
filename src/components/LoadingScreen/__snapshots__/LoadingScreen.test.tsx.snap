// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading screen markup 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-17nwl61-Wrapper-mixinFlexCenter e1ozef230",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-1ov65rm-Text e1ozef231",
        },
        "children": Array [
          Object {
            "data": "Loading",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "css-4uzb0-ProgressCircle",
            "data-test-id": "container",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "css-15gjpuu-ProgressCircle",
                "data-test-id": "svg",
                "height": "20",
                "width": "20",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "css-11b015i-ProgressCircle",
                    "cx": "10",
                    "cy": "10",
                    "data-test-id": "circle-1",
                    "r": "9",
                    "stroke": "pink",
                    "stroke-width": "2",
                  },
                  "children": Array [],
                  "name": "circle",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": Object {
                    "attribs": Object {
                      "class": "svg-circle css-wqmigj-ProgressCircle-ProgressCircle",
                      "cx": "10",
                      "cy": "10",
                      "data-test-id": "circle-2",
                      "r": "9",
                      "stroke": "red",
                      "stroke-dasharray": "56.548667764616276",
                      "stroke-dashoffset": "56.548667764616276",
                      "stroke-width": "2",
                    },
                    "children": Array [],
                    "name": "circle",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "cx": undefined,
                      "cy": undefined,
                      "data-test-id": undefined,
                      "r": undefined,
                      "stroke": undefined,
                      "stroke-dasharray": undefined,
                      "stroke-dashoffset": undefined,
                      "stroke-width": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "cx": undefined,
                      "cy": undefined,
                      "data-test-id": undefined,
                      "r": undefined,
                      "stroke": undefined,
                      "stroke-dasharray": undefined,
                      "stroke-dashoffset": undefined,
                      "stroke-width": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "svg-circle css-wqmigj-ProgressCircle-ProgressCircle",
                    "cx": "10",
                    "cy": "10",
                    "data-test-id": "circle-2",
                    "r": "9",
                    "stroke": "red",
                    "stroke-dasharray": "56.548667764616276",
                    "stroke-dashoffset": "56.548667764616276",
                    "stroke-width": "2",
                  },
                  "children": Array [],
                  "name": "circle",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "css-11b015i-ProgressCircle",
                      "cx": "10",
                      "cy": "10",
                      "data-test-id": "circle-1",
                      "r": "9",
                      "stroke": "pink",
                      "stroke-width": "2",
                    },
                    "children": Array [],
                    "name": "circle",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "cx": undefined,
                      "cy": undefined,
                      "data-test-id": undefined,
                      "r": undefined,
                      "stroke": undefined,
                      "stroke-width": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "cx": undefined,
                      "cy": undefined,
                      "data-test-id": undefined,
                      "r": undefined,
                      "stroke": undefined,
                      "stroke-width": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-dasharray": undefined,
                    "stroke-dashoffset": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-dasharray": undefined,
                    "stroke-dashoffset": undefined,
                    "stroke-width": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-test-id": undefined,
                "height": undefined,
                "width": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-test-id": undefined,
                "height": undefined,
                "width": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-test-id": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "css-4uzb0-ProgressCircle",
          "data-test-id": "container",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-15gjpuu-ProgressCircle",
              "data-test-id": "svg",
              "height": "20",
              "width": "20",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-11b015i-ProgressCircle",
                  "cx": "10",
                  "cy": "10",
                  "data-test-id": "circle-1",
                  "r": "9",
                  "stroke": "pink",
                  "stroke-width": "2",
                },
                "children": Array [],
                "name": "circle",
                "namespace": "http://www.w3.org/2000/svg",
                "next": Object {
                  "attribs": Object {
                    "class": "svg-circle css-wqmigj-ProgressCircle-ProgressCircle",
                    "cx": "10",
                    "cy": "10",
                    "data-test-id": "circle-2",
                    "r": "9",
                    "stroke": "red",
                    "stroke-dasharray": "56.548667764616276",
                    "stroke-dashoffset": "56.548667764616276",
                    "stroke-width": "2",
                  },
                  "children": Array [],
                  "name": "circle",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-dasharray": undefined,
                    "stroke-dashoffset": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-dasharray": undefined,
                    "stroke-dashoffset": undefined,
                    "stroke-width": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "data-test-id": undefined,
                  "r": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "data-test-id": undefined,
                  "r": undefined,
                  "stroke": undefined,
                  "stroke-width": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "svg-circle css-wqmigj-ProgressCircle-ProgressCircle",
                  "cx": "10",
                  "cy": "10",
                  "data-test-id": "circle-2",
                  "r": "9",
                  "stroke": "red",
                  "stroke-dasharray": "56.548667764616276",
                  "stroke-dashoffset": "56.548667764616276",
                  "stroke-width": "2",
                },
                "children": Array [],
                "name": "circle",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "css-11b015i-ProgressCircle",
                    "cx": "10",
                    "cy": "10",
                    "data-test-id": "circle-1",
                    "r": "9",
                    "stroke": "pink",
                    "stroke-width": "2",
                  },
                  "children": Array [],
                  "name": "circle",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "data-test-id": undefined,
                    "r": undefined,
                    "stroke": undefined,
                    "stroke-width": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "data-test-id": undefined,
                  "r": undefined,
                  "stroke": undefined,
                  "stroke-dasharray": undefined,
                  "stroke-dashoffset": undefined,
                  "stroke-width": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "data-test-id": undefined,
                  "r": undefined,
                  "stroke": undefined,
                  "stroke-dasharray": undefined,
                  "stroke-dashoffset": undefined,
                  "stroke-width": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-test-id": undefined,
              "height": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-test-id": undefined,
              "height": undefined,
              "width": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "css-1ov65rm-Text e1ozef231",
          },
          "children": Array [
            Object {
              "data": "Loading",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-test-id": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
